src/Symfony/Component/RateLimiter/LimiterFactory.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\RateLimiter\Storage\InMemoryStorage;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class LimiterFactory
```

Comment:
I'm not sure we need such a class.

Reply:
Hmm, I think we need such a class for integration in Symfony. I would love to allow users to configure limiters in the config, e.g:

```yaml
framework:
  limiter:
    api:
      storage: ...
      rate: { hour: 10 }
      initial: 10
```

Based on this limiter, a specific limiter with an id should be created (e.g. client IP for limiter ids). I.e:

```php
public function someApiAction(LimiterFactory $factory, Request $request)
{
    $limiter = $factory->createLimiter('api', $request->getClientId());
    if (!$limiter->consume()) {
        return new JsonResponse(0, 429);
    }

   // ...
}
```